diff --git a/pkgs/development/compilers/llvm/common/libunwind/default.nix b/pkgs/development/compilers/llvm/common/libunwind/default.nix
index bbd33110149f..ff43f945e771 100644
--- a/pkgs/development/compilers/llvm/common/libunwind/default.nix
+++ b/pkgs/development/compilers/llvm/common/libunwind/default.nix
@@ -51,6 +51,8 @@ stdenv.mkDerivation (finalAttrs: {
     python3
   ];

+  env.LDFLAGS = "-unwindlib=none";
+
   cmakeFlags = [
     (lib.cmakeBool "LIBUNWIND_ENABLE_SHARED" enableShared)
     (lib.cmakeFeature "LLVM_ENABLE_RUNTIMES" "libunwind")
